src/bindings/c/ov/src/ov_c_api.cpp

```diff
@@ -613,47 +618,91 @@ bool ov_model_is_dynamic(const ov_model_t* model) {
 
 std::vector<char*> split(std::string s) {
```

Comment:
I suggest to make this helper function look like this. More verbose naming to self-explain what the function does. Also you can return a vector of strings instead because at the call site you're currently converting those const char* values to strings anyway.
```suggestion
std::vector<std::string> split_to_dims(const std::string& partial_shape) {
```

Reply:
I accept, a readable name is friendly to reader, and use std::string is more straightforward.